### PR TITLE
Feature/dcc 5774 dcc repository investigate indexing song supported repositories

### PR DESCRIPTION
**Pull request by KevinFHartmann**
This is ready for a PR, assuming that the output is what we want it to be. :-)

It looks pretty good to me, but I might have missed/misunderstood something, so it would probably be good to have someone else review it, too. :-)

### CODE REVIEW

File: dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+@RequiredArgsConstructor
+public class SongProcessor {
+    @NonNull
+    private final Repository repository;
+    @NonNull
+    private final RepositoryFileContext context;
+
+    public Iterable<RepositoryFile> getRepositoryFiles(Iterable<SongAnalysis> analyses) {
+        return StreamSupport.stream(analyses.spliterator(), false).
```
Comment (andricDu) [inline dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java]: Method reference and static imports

```java
        return stream(analyses.spliterator(), false).
                flatMap(this::convertFiles).collect(toList());
```

File: dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                flatMap(analysis->convertFiles(analysis)).collect(Collectors.toList());
+    }
+
+    public Stream<RepositoryFile> convertFiles(SongAnalysis analysis) {
+        val results = new ArrayList<RepositoryFile>();
+
+        for(val file: analysis.getFiles()) {
```
Comment (andricDu) [inline dcc-repository-song/src/main/java/org/icgc/dcc/repository/song/core/SongProcessor.java]: This whole method could be shortened down to:

```java
        return analysis.getFiles().stream()
            .filter(f -> !isIndexFile(f.get(fileName)))
            .map(f -> convert(f, analysis));
```

**Review by andricDu — CHANGES_REQUESTED** [review]
I've noticed some of the files are using 4 spaces for indentation. Lets bring them back to 2 for consistency with the rest of our code base. 

Once you set the default in IntelliJ you can just use the IDE to reformat.

**Comment by KevinFHartmann** [comment]
This is ready to be reviewed again. :-)

**Review by andricDu — APPROVED** [review]

File: dcc-repository-core/src/main/java/org/icgc/dcc/repository/core/util/RepositoryFiles.java [hunk 3]
```diff
@@ -40,7 +41,16 @@ public static boolean isPCAWGFile(@NonNull RepositoryFile file) {
   }
 
   public static boolean isPCAWGFileCopy(@NonNull FileCopy fileCopy) {
-    return fileCopy.getRepoOrg().equals(PCAWG.getId());
+    if (fileCopy == null) {
```
Comment (andricDu) [inline dcc-repository-core/src/main/java/org/icgc/dcc/repository/core/util/RepositoryFiles.java]: fileCopy cannot be null due to the `@NonNull`

**Review by andricDu — CHANGES_REQUESTED** [review]

**Review by andricDu — APPROVED** [review]

**Pull request merged**